wagtail_newsletter/panels.py

```diff
@@ -18,6 +19,24 @@ class BoundPanel(Panel.BoundPanel):
 
         instance: "models.NewsletterPageMixin"
 
+        class Media:
+            js = [
+                "wagtail_newsletter/js/wagtail_newsletter.js",
+            ]
+
+        @cached_property
+        def permissions(self):
+            return frozenset(
+                action
+                for action in [
```

Comment:
Since we're using micro-optimisations such as using `frozenset`... Maybe we could go "full on" on this, and use a tuple for these actions rather than a list? :smile:

Reply:
Ah, that wasn't a micro-optimisation, just a bit of paranoia that somebody might mutate the set, not realising that it's a _cached_ property.

There's this idea that tuples should be used when the position of each item matters, e.g. in function `*args`, while lists should be used when it's an enumeration of similar kinds of values. I can't find a great source, but https://stackoverflow.com/questions/19733666/what-is-homogenous-in-python-list-documentation has some useful thoughts.